### ISSUE TITLE
StrawberrryShake Add Overload to ExecuteAsync to allow extensions to be passed. 

### ISSUE THREAD

**Issue by MattMinke** [ChilliCream/graphql-platform#5204]
### Is your feature request related to a problem?

A 3rd party API we are integrating with requires us to add additional headers to each request that are unique to the request and are not known when the client is registered with the service collection. Combining the suggested solution below with a custom IHttpConnection implementation that reads from the OperationRequest.Extensions will allow us to satisfy the behavior required by the 3rd party. 

We can replace the IHttpConnection using this code. 

```C#

branchClientBuilder.ClientServices.Replace<IHttpConnection>(ServiceLifetime.Singleton, (provider =>
{
    var clientFactory = provider.GetRequiredService<global::System.Net.Http.IHttpClientFactory>();
        return new MyCustomHttpConnection(() => clientFactory.CreateClient("BranchClient"));
}));
```

### The solution you'd like

Add an overload to ExecuteAsync for each query that allows an extensions collection to be added to the OperationRequest 

![image](https://user-images.githubusercontent.com/5995359/176789742-e089402f-e802-4f0f-8da5-1de324ec6b31.png)

### Product

Strawberry Shake

**Comment by PascalSenn** [comment]
@MattMinke  you know that you can configure the http client factory? you can also configure the http client builder as the third argument of `ConfigureHttpClient`
https://chillicream.com/docs/strawberryshake/networking/authentication#http

**Comment by MattMinke** [comment]
Yep I am aware of that. unfortunately It does not solve the issue in our case. The value we need to add to the headers is unique per request. The client builder will let us add a header at configuration time but that is too soon. The header changes with each request and is not consistent. The header has to be calculated at request time. 

With the solution I am proposing we would be able to add additional values to the request.Extensions dictionary and then access this dictionary in a custom IHttpConnection.

**Comment by mac2000** [comment]
@jorrit may be simplified example from #3467 [comment](https://github.com/ChilliCream/hotchocolate/issues/3467#issuecomment-942017053) should be added to [documentation page](https://chillicream.com/docs/strawberryshake/networking/authentication)

it might solve many future issues

in our case we have console kafka consumer which uses strawberry shake client so we have no http context to pass headers from but example above did work perfectly

**Comment by MattMinke** [comment]
This issue is currently marked as part of milestone [HC-13.0.0](https://github.com/ChilliCream/graphql-platform/milestone/65). Please do not close this issue. I have been waiting patiently for this milestone to be complete so I can take advantage of this.

**Comment by danny-zegel-zocdoc** [comment]
as I commented in https://github.com/ChilliCream/graphql-platform/issues/6426 I would also appreciate the ability to specify custom headers per request.

**Comment by sabbadino** [comment]
> This issue is currently marked as part of milestone [HC-13.0.0](https://github.com/ChilliCream/graphql-platform/milestone/65). Please do not close this issue. I have been waiting patiently for this milestone to be complete so I can take advantage of this.
I need to pass per request custom headers 
milestone [HC-13.0.0] is closed .. was it done ? if yes , how do I pass per request custom headers ?

**Comment by michaelstaib** [comment]
This is now possible with 15.1.5 ... have a look at my comment here: https://github.com/ChilliCream/graphql-platform/issues/3467
